### PR TITLE
Use state updater pattern

### PR DESCRIPTION
**Pull request by jaredpalmer**
Whenever you need to adjust state based on previous state, better to use the updater/reducer pattern of setState.

### CODE REVIEW

File: packages/react-instantsearch/examples/geo-search/src/App.js [hunk 1]
```diff
@@ -32,10 +32,14 @@ class App extends Component {
 
   // when a click on the map is performed, the query is reset and the aroundLatLng property updated.
   onLatLngChange({ lat, lng }) {
-    this.setState({
-      searchState: { ...this.state.searchState, query: '' },
+    this.setState(prevState => ({
+      ...prevState,
```
Comment (Haroenv) [inline packages/react-instantsearch/examples/geo-search/src/App.js]: I thought the updater did a shallow merge, so there’s no need to spread

File: packages/react-instantsearch/examples/geo-search/src/App.js [hunk 2]
```diff
@@ -32,10 +32,14 @@ class App extends Component {
...
+      searchState: {
+        ...prevState.searchState, 
+        query: ''
```
Comment (Haroenv) [inline packages/react-instantsearch/examples/geo-search/src/App.js]: Can you add a trailing comma so prettier is happy?

**Comment by vvo** [comment]
I wonder if:
1. this can trigger different behaviour for our users
2. it can actually fix some bad behaviours we had recently (infinite loops, double requests)

cc @samouss

**Comment by Haroenv** [comment]
I think this will fix 2 indeed (no out-of-order updates), but this is just the example, we need it in some code too I think @vvo

**Comment by jaredpalmer** [comment]
😎

**Comment by samouss** [comment]
> 1. this can trigger different behaviour for our users

Even it's only in the example, the impact on the users should be null.

> 2. it can actually fix some bad behaviours we had recently (infinite loops, double requests)

I don't think so, not yet investigate this behaviours.

**Pull request merged**